proposals/2023/05-26_vela_ui_rewrite.md

```diff
@@ -0,0 +1,217 @@
+  - `gofmt` alike, automatic formatting without debate
+- [eslint](https://eslint.org/)
+  - `golangci-lint` alike, automatic linting, recommending best coding practices
+
+<details>
+<summary><b>Why react?</b></summary>
+
+React has been a staple frontend "framework" in 2023. It has wide adoption and has a massive practitioner base. In the [2023 Stackoverflow Survey](https://survey.stackoverflow.co/2023/#section-admired-and-desired-web-frameworks-and-technologies), it continues to earn top a top position in desirability and admiration.
+
+We think that React, among others "frameworks" like Angular, Vue or Svelte fulfills our desire to increase casual contributors in this codebase. The principle reason is the sheer momentum that React has.
+
+React, like most frontend tooling, will enter its halcyon days and the industry will move to new tooling. There is no such thing as future-proof. However, in an effort to be future-resistant we are specifically making choices that will be easy to change around build tooling. We are not entirely leveraging the React core team's ecosystem (Next, CRA, Jest), and instead using a broader community ecosystem (Vite).
+
+</details>
+
+<details>
+<summary><b>Can we achieve the same and better results than Elm with this new approach?</b></summary>
+
+Elm offered a bundled developer experience for webapps. The React-centric ecosystem is much more fragmented, but it lets you pick from a variety of options that are easier to plug in and to swap as needs change.
+
+</details>
+
+We have considered these functional changes:
+
+- new/updated endpoints or url paths
+  - re-evaluating the `/-/secrets` path
+  - re-evaluating paths that shadow potentially valid Organization, Repository, or Secret names
+- new/updated configuration variables (environment, flags, files, etc.)
+  - environment variables for features (e.g. default pull request settings, auto-linkifying in logs, etc)
+
+From an overall perspective, the goal is to:
+
+- Replicate the current user interface (look'n'feel; design) in the technology stack listed above as closely as possible
+- Replicate the current functionality in the technology stack listed above as closely as possible
+
+Consider this redesign and rewrite a `1-to-1.3̅3̅3̅` port rather than a `1-to-1` port.
+
+Further enhancements can and will be made after.
+
+## Implementation
+
+<!--
+This section is intended to explain how the solution will be implemented for the proposal.
+
+NOTE: If there are no current plans for implementation, please leave this section blank.
+-->
+
+- Since the UI is a different repo and there are no co-dependency's the new UI should be a drop-in replacement and in theory should be able to run in parallel to the existing version if necessary.
+
+**Please briefly answer the following questions:**
+
+1. Is this something you plan to implement yourself?
+
+Yes. There is a small team working on this (see below; hackathon activity).
+
+2. What's the estimated time to completion?
```

Comment:
This proposal was completed prior to the hackathon. Post-hackathon, can you provide more details and refine what it would take to complete this, how many people are committed, and their timeline to completion?

Reply:
- Ryan Rampersad (me)
- Zac Skalko @zapplebee 

Neither of us are working on this full-time. I worked on this for fun because I use it for work regularly and I think it's a great tool. I want to add features. I want it to be approachable by more people. I want more people to use it.

I would hope we are not replacing two primary contributors with an alternative set of two primary contributors. It's not a good exchange. It's not our intent. Hopefully this makes it easier for other contributors on the existing teams and in the community to join in.

In our hackathon ui repo, we have an overview in the README that shows a majority outstanding tasks. We can add more detail. We have not updated this lately due to some internal feedback on this topic. Once we can align this proposal better, we can estimate and allocate time better.

James mentioned 30-60 days in the document. We will refine that estimate when the time comes.

Reply:
> Neither of us are working on this full-time

I love the enthusiasm of individuals wanting to contribute. Naturally, when we hear that individuals are interested in rewriting "this for fun" makes us a little concerned about the overall commitment. This is a full-time job for us and our talents are spread across the entire codebase - we're not structured where we have two people working on the front-end and we'd just need to retrain those individuals on React. This is a whole-team effort for us - which is part of why we're taking this decision very seriously.